language/analyze.js

```diff
@@ -12,163 +12,294 @@
-    } else {
+    console.log('Done analyzing syntax');
+    return callback(null, apiResponse);
+  });
+}
+// [END analyze_syntax]
+
+// [START analyze_syntax_from_file]
+/**
+ * Detect the syntax in a text file that resides in Google Cloud Storage.
+ *
+ * @param {string} bucket The bucket where the file resides.
+ * @param {string} filename The name of the file to be analyzed.
+ * @param {object} [options] Optional configuration.
+ * @param {string} [options.language] The language of the text, e.g. "en".
+ * @param {string} [options.type] "text" or "html".
+ * @param {function} callback The callback function.
+ */
+function analyzeSyntaxFromFile (bucket, filename, options, callback) {
+  var document = language.document({
+    content: storage.bucket(bucket).file(filename),
+    type: options.type,
+    language: options.language
+  });
+
+  var config = {
+    syntax: true
+  };
+
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/language/latest/language/document
+  document.annotate(config, function (err, result, apiResponse) {
+    if (err) {
       return callback(err);
     }
+
+    console.log('Done analyzing syntax');
+    return callback(null, apiResponse);
   });
-};
+}
+// [END analyze_syntax_from_file]
+// [END all]
 
-if (require.main === module) {
-  var args = process.argv.slice(2);
-  if (args.length !== 2) {
-    console.log('Incorrect number of arguments. ' +
-                'Usage: node analyze.js <sentiment|entities|syntax> <text>');
-    process.exit(1);
-  }
-  if (['sentiment', 'entities', 'syntax'].indexOf(args[0]) === -1) {
-    console.log('Incorrect command. ' +
-                'Usage: node analyze.js <sentiment|entities|syntax> <text>');
-    process.exit(1);
+// The command-line program
+var cli = require('yargs');
+var utils = require('../utils');
+
+var program = module.exports = {
+  analyzeSentiment: analyzeSentiment,
+  analyzeSentimentFromFile: analyzeSentimentFromFile,
+  analyzeEntities: analyzeEntities,
+  analyzeEntitiesFromFile: analyzeEntitiesFromFile,
+  analyzeSyntax: analyzeSyntax,
+  analyzeSyntaxFromFile: analyzeSyntaxFromFile,
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
   }
-  exports.main(args[0], args[1], function (err, result) {
-    if (err) {
-      console.error(err);
+};
+
+cli
+  .demand(1)
+  .command('sentiment <text>', 'Detect the sentiment of a block of text.', {}, function (options) {
+    program.analyzeSentiment(options.text, utils.pick(options, ['language', 'type']), utils.makeHandler());
+  })
+  .command('sentimentFromFile <bucket> <filename>', 'Detect the sentiment of text in a GCS file.', {}, function (options) {
+    program.analyzeSentimentFromFile(options.bucket, options.filename, utils.pick(options, ['language', 'type']), utils.makeHandler());
+  })
+  .command('entities <text>', 'Detect the entities of a block of text.', {}, function (options) {
```

Comment:
I feel like these should be something like "entitiesFromText" or something like that. "Entities" gives me the impression that "entitiesFromText" is a conceptual subclass of this, while to me they appear to be on the same level conceptually.

Thoughts?